advanced/pihole-FTL.systemd

```diff
@@ -20,13 +19,19 @@ PermissionsStartOnly=true
 Type=forking
 PIDFile=/run/pihole-FTL.pid
 
-ExecStartPre=/bin/bash /opt/pihole/pihole-FTL-prestart.sh
-;ExecStartPre=/sbin/setcap cap_net_bind_service,cap_net_raw,cap_net_admin+eip /usr/bin/pihole-FTL
+ExecStartPre=/bin/touch /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
+ExecStartPre=/bin/mkdir -p /var/run/pihole /var/log/pihole
+ExecStartPre=/bin/chown pihole:pihole /var/run/pihole /var/log/pihole
+ExecStartPre=-/bin/rm /var/run/pihole/FTL.sock
+ExecStartPre=/bin/chown pihole:pihole /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /etc/pihole /etc/pihole/dhcp.leases /var/log/pihole.log
+ExecStartPre=/bin/chmod 0644 /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
```

Comment:
Wouldn't it be much simpler, if there was a single /var/log/pihole/ directory. This directory would be created with the proper permissions, so the unprivileged pihole-FTL process can write into it. It should then be sufficient to create this directory once during installation.
As for the files in /run, having a /run/pihole directory with the proper permissions where all runtime files are stored would be much simpler as well. This directory could be managed by systemd directly via
```RuntimeDirectory=```. systemd would take care of creating that directory, chowning it, and cleaning it up when the service is stopped.

Reply:
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=

Reply:
I'd actually made the assumption that all the setup stuff that get's done every time pihole starts is to make up for the fact that it's not actually "installed" per se, it's just a script that needs to do it's own set up every time.

So it can never assume that the relevant directories exist nor that the relevant caps are set on binaries or that anything has been set up at all really, hence the need to do so much setup as root then try and drop privs down to a non-root user. I mean, might as well make this thing compilable in some language to get a binary and throw CAP_SYS_ADMIN onto it... just seems like a bunch of hacks which seem to work for now because conventional approaches to doing things seem to have all gone out the window 

If it where possible ideally there would be .deb .rpm .whatever packages that could handle the creation of /var/log/pihole /var/run/pihole /etc/pihole /whateverelsewe need as well as handing any permission munging or, alternatively, have the process run as a proper executable binary that can depermission itself once it's done its thing (similar to dnsmasq, runs as root but the process very clearly drops most of the privileges it has pretty quickly if you strace a startup).

It feels like making pihole work with systemd means making pihole work as an actual linux service that follows all associated conventions, which is a hell of an overhaul.

Reply:
That's a fair statement, and if you wanted to open a new Feature Request on https://discourse.pi-hole.net asking for the inclusion of a systemd service file in the pi-hole/pi-hole repo, I think that we most likely would be able to satisfy it.